ci-operator/step-registry/cucushift/installer/check/gcp/credentials-mode/cucushift-installer-check-gcp-credentials-mode-commands.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ret=0
+
+echo "INFO: Checking cluster credentials mode..."
+
+curr_credentials_mode=$(oc get cloudcredential cluster -o json | jq -r ".spec.credentialsMode")
+if [ -z "${curr_credentials_mode}" ]; then
+    curr_credentials_mode="Mint"
+fi
+echo "INFO: Current credentials mode: '${curr_credentials_mode}'"
+
+expected_credentials_mode=""
+if [ -n "${CREDENTIALS_MODE}" ]; then
+    expected_credentials_mode="${CREDENTIALS_MODE}"
+elif test -f "${SHARED_DIR}/install-config.yaml"; then
+    expected_credentials_mode=$(yq-go r "${SHARED_DIR}/install-config.yaml" credentialsMode)
+fi
+if [ -z "${expected_credentials_mode}" ]; then
+    expected_credentials_mode="${curr_credentials_mode}"
```

Comment:
If `expected_credentials_mode` is null, shall we set expected_credentials_mode to the default credentials mode "Mint" instead?
`expected_credentials_mode="Mint"`

Reply:
_FYI_
- The check aims for non-Mint mode, i.e. Passthrough mode and Manual mode.
- Line 43, along with line 32, is sort of fault tolerance, and expecting the check would pass in such case.